// Copyright 2025 gRPC authors.

syntax = "proto3";

package chaotic_good_frame;

import "src/core/ext/transport/chaotic_good/chaotic_good_frame.proto";

message MessageFrame {
  uint32 stream_id = 1;
  oneof type {
    bytes data = 2;
    uint32 all_zeros_length = 3;
  }
}

message ClientMetadataFrame {
  uint32 stream_id = 1;
  ClientMetadata payload = 2;
}

message ServerMetadataFrame {
  uint32 stream_id = 1;
  ServerMetadata payload = 2;
}

message BeginMessageFrame {
  uint32 stream_id = 1;
  BeginMessage payload = 2;
}

message EmptyStreamFrame {
  uint32 stream_id = 1;
}

message TestFrame {
  oneof type {
    Settings settings = 1;
    ClientMetadataFrame client_initial_metadata = 2;
    ServerMetadataFrame server_initial_metadata = 3;
    ServerMetadataFrame server_trailing_metadata = 4;
    MessageFrame message = 5;
    MessageFrame message_chunk = 6;
    BeginMessageFrame begin_message = 7;
    EmptyStreamFrame client_end_of_stream = 8;
    EmptyStreamFrame cancel = 9;
  }
}
